feat(client-utils): map Stellar trustline keyring transactions (#9440)
MetaMask/core: packages/client-utils/CHANGELOG.md

  - `mapKeyringTransaction` for mapping keyring transactions to activity items
  - `mapLocalTransaction` for mapping local transaction groups to activity items
  - Shared activity types (`ActivityItem`, `ActivityKind`, `Status`, etc.)

> - Add `assetActivation`/`assetDeactivation` activity kinds and map Stellar trustline `TokenApprove`/`TokenDisapprove` keyring transactions to them in `mapKeyringTransaction` ([#9440](https://github.com/MetaMask/core/pull/9440))

Reviewer: ```suggestion
- Map `assetActivation` and `assetDeactivation` activity types in transaction activity mappers ([#9440](https://github.com/MetaMask/core/pull/9440))
```
